src/git/RepositoryManager.ts

```diff
@@ -116,6 +144,60 @@ export class RepositoryManager implements vscode.Disposable {
     }
   }
 
+  private async handleBranchChange() {
+    // update pull requests to be up to date
+    await this.refreshPullRequests()
+
+    this._branch = this._current?.state.HEAD?.name
+
+    if (!this._branch) {
+      Logger.warn(`No HEAD information found: ${JSON.stringify(this._current?.state.HEAD)}`)
+      this.prState = PullRequestState.NoPullRequest
+      this.branchState = BranchState.OnUnknownBranch
+    }
+
+    // in which kind of branch are we? (PR, analysed, unknown)
+    else if (this._enabledBranches.some((b) => b.name === this._branch)) {
+      Logger.appendLine(
+        `Current branch is an analysed branch: ${this._branch}. Skipped looking for a pull request. Loading branch issues...`
```

Comment:
We'll need to update these with American English (analyzed vs. analysed), but we can do this in a later dedicated PR.